polkadot/xcm/src/lib.rs

```diff
@@ -193,68 +224,59 @@ macro_rules! versioned_type {
 		)]
 		#[codec(encode_bound())]
 		#[codec(decode_bound())]
+        $(#[scale_info(bounds(), skip_type_params($($gen)+))])?
 		#[scale_info(replace_segment("staging_xcm", "xcm"))]
 		$(#[$attr])*
-		pub enum $n {
+		pub enum $n $(<$($gen),+>)? {
 			$(#[$index2])*
 			V2($v2),
 			$(#[$index3])*
 			V3($v3),
 			$(#[$index4])*
 			V4($v4),
 		}
-		impl $n {
+		impl$(<$($gen),+>)? $n $(<$($gen),+>)? {
```

Comment:
```suggestion
		impl<$($gen),+> $n <$($gen),+> {
```

Rust is fine to have empty `<>`.